hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/DMROperationParamAttributes.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.agent.monitor.extension;
+
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.registry.AttributeAccess;
+import org.jboss.dmr.ModelType;
+
+public interface DMROperationParamAttributes {
+
+
+    SimpleAttributeDefinition TYPE = new SimpleAttributeDefinitionBuilder("type",
+            ModelType.STRING)
+                    .setAllowNull(false)
```

Comment:
in the .xsd, you have:
+    <xs:attribute name="type"         type="paramTypeType" use="optional" default="string"/>

if the default is "string" we need to specify that here via:

   .setDefaultValue(new ModelNode("string"))